eegnb/experiments/Experiment.py

```diff
@@ -0,0 +1,100 @@
+""" 
+Initial run of the Experiment Class Refactor base class
+
+Derived classes have to set a few things in major:
+1. load_stimulus function : returns an array of stimuli
+2. present_stimulus function : presents the stimuli and pushes eeg data back and forth as needed
+Additional parameters can be set from the derived class as per the initializer
+
+"""
+
+class Experiment:
+
+    def __init_(self, exp_name):
+        """ Anything that must be passed as a minimum for the experiment should be initialized here """
+        
+        """ Dk if this overwrites the class variable or is worth doing 
+        if we just assume they will overwrite """
+        
+        self.exp_name= exp_name
+        self.instruction_text = """\nWelcome to the {} experiment!\nStay still, focus on the centre of the screen, and try not to blink. \nThis block will run for %s seconds.\n
+        Press spacebar to continue. \n""".format(exp_name) 
+        self.duration=120 
+        self.eeg:EEG=None 
+        self.save_fn=None 
+        self.n_trials=2010
+        self.iti = 0.4
+        self.soa = 0.3
```

Comment:
@JohnGriffiths Always wondered about these confusing variable names. Are `iti` and `soa` abbreviations or something?

In any case, they should probably be given more descriptive names (and note that the values are different in different experiments, so they need to be passed too!)

Reply:
yeah so they are being initialized to default values and then if the experiment needs different ones it can change it after the init (otherwise I could force it to initialize too) - see visual ssvep

Reply:
They are standard technical acronyms in psych / cog neuro: 

ITI = [Inter-trial interval](https://www.psychstudio.com/articles/intertrial-interval/#:~:text=Intertrial%20Interval%20is%20defined%20as,onset%20of%20the%20next%20trial)

SOA = [Stimulus onset asynchrony](https://en.wikipedia.org/wiki/Stimulus_onset_asynchrony)

( A related term that we don't have here is ISI (inter-stimulus interval) )

These absolutely should be more clearly defined and consistently used.